torch/linalg/__init__.py

```diff
@@ -140,6 +140,111 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+svd = _add_docstr(_linalg.linalg_svd, r"""
+linalg.svd(input, full_matrices=True, compute_uv=True, out=None) -> (Tensor, Tensor, Tensor)
+
+This function returns a namedtuple ``(U, S, Vh)`` which is the singular value
+decomposition of a input real matrix or batches of real matrices :attr:`input` such that
+:math:`input = U \times diag(S) \times V^H` (where :math:`V^H` is ``Vh``).
+
+.. warning:: **Differences with** :meth:`~torch.svd`:
+
+             * :attr:`full_matrices` is the opposite of
+               :meth:`~torch.svd`'s :attr:`some`. Note that default value
+               for both is ``True``, so the default behavior is effectively
+               the opposite.
+
+             * it returns ``Vh``, whereas :meth:`~torch.svd` returns
+               ``V``. The result is that when using :meth:`~torch.svd` you
+               need to manually transpose and conjugate ``V`` in order to
+               reconstruct the original matrix.
+
+             * If :attr:`compute_uv=False`, it returns empty tensors (i.e.,
+               with 0 elements) for ``U`` and ``V``, whereas
+               :meth:`~torch.svd` returns zero-filled tensors.
+
+             **Differences with** ``numpy.linalg.svd``:
+
+             * if :attr:`compute_uv=False` it returns ``(empty_tensor, S, empty_tensor)``,
+               whereas numpy returns ``S``.
+
+
+The dtype of ``U`` and ``V`` is the same as the ``input`` matrix. The dtype of
+``S`` is always real numbers, even if ``input`` is complex.
+
+If :attr:`full_matrices` is ``False``, the method returns the reduced singular value decomposition
+i.e., if the last two dimensions of :attr:`input` are ``m`` and ``n``, then the returned
+`U` and `V` matrices will contain only :math:`min(n, m)` orthonormal columns.
+
+If :attr:`compute_uv` is ``False``, the returned `U` and `V` will be None.:attr:`full_matrices` will
+be ignored here.
+
+.. note:: The singular values are returned in descending order. If :attr:`input` is a batch of matrices,
+          then the singular values of each matrix in the batch is returned in descending order.
+
+.. note:: The implementation of SVD on CPU uses the LAPACK routine `?gesdd` (a divide-and-conquer
+          algorithm) instead of `?gesvd` for speed. Analogously, the SVD on GPU uses the MAGMA routine
+          `gesdd` as well.
```

Comment:
This note could be:

"PyTorch's implementation of SVD uses LAPACK's (on CPU) or MAGMA's (on CUDA) \`?gesdd\` (a divide-and-conquer algorithm) instead of `?gesvd` for speed.

What's up with the question marks before gesdd and gesvd?

Reply:
Just for the information.
MKL documentation for LAPACK uses the question marks in place of a letter indicating what datatype the function operates on (`{'s','d','c','z'} -> '?'`).
https://software.intel.com/content/www/us/en/develop/documentation/mkl-developer-reference-fortran/top/lapack-routines/naming-conventions-for-lapack-routines.html
Netlib documentation for LAPACK the first datatype letter is replaced with 'x'
https://www.netlib.org/lapack/lug/node24.html

Reply:
Oh I see. I suppose we could remove the question mark or replace it with the types we support in braces, like: {a, b}gesdd, but the current presentation also seems fine.

Reply:
A quick search in the docs seems to suggest that we are not using a consistent naming scheme. E.g.:

- `cholesky_inverse` explicitly lists all variations by saying ``dpotri and spotri``

- `toch.geqrf` mentions both `geqrf` and `?geqrf` in the same docstring

- `orgqr` the same

Reply:
It's not surprising we're inconsistent, and since we're inconsistent this PR can do whatever you like.

cc @heitorschueroff, too. We should review the linear algebra documentation for consistency ahead of the 1.8 release.